src/cdk/a11y/focus-trap.ts

```diff
@@ -321,23 +321,53 @@ export class FocusTrapDeprecatedDirective implements OnDestroy, AfterContentInit
   selector: '[cdkTrapFocus]',
   exportAs: 'cdkTrapFocus',
 })
-export class FocusTrapDirective implements OnDestroy, AfterContentInit {
+export class CdkTrapFocus implements OnDestroy, AfterContentInit {
   focusTrap: FocusTrap;
 
+  /** Previously focused element to restore focus to upon destroy when using autoCapture. */
+  private _previouslyFocusedElement: HTMLElement | null = null;
+
   /** Whether the focus trap is active. */
   @Input('cdkTrapFocus')
   get enabled(): boolean { return this.focusTrap.enabled; }
   set enabled(value: boolean) { this.focusTrap.enabled = coerceBooleanProperty(value); }
 
-  constructor(private _elementRef: ElementRef, private _focusTrapFactory: FocusTrapFactory) {
+  /**
+   * Whether the directive should automatially move focus into the trapped region upon
+   * initialization and return focus to the previous activeElement upon destruction.
+   */
+  @Input('cdkTrapFocusAutoCapture')
+  get autoCapture(): boolean { return this._autoCapture; }
```

Comment:
Since this won't react to value changes, maybe it should be a plain `@Input` without the coercion or it should take the value via an `@Attribute`?

Reply:
I made it an input since people may want to build a higher-level component on top of it and forward from their own API for the equivalent behavior

Reply:
Makes sense, although I do feel like we're starting to add the coercion to everything.

Reply:
I'm going to push for Angular to add `@BooleanInput` (and `@NumberInput`) to help with this.